core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala

```diff
@@ -87,6 +87,8 @@ class GroupCoordinator(val brokerId: Int,
 
   private val isActive = new AtomicBoolean(false)
 
+  val epochForPartitionId = mutable.Map[Int, Int]()
```

Comment:
Does this need to be a concurrent collection? It does not look like we can count on a lock protecting `onElection` and `onResignation`.

Reply:
Yes it does, thanks! Now fixed.